File: frontend/src/types/ask.ts
export type AskPhase =
  | 'idle'
  | 'retrieving'
  | 'confirming'
  | 'generating'
  | 'answered'
  | 'insufficient'
  | 'empty'
  | 'error'
  | 'cancelled';

export interface AskRequest {
  question: string;
  meetingIds: string[];
  dateFrom: string | null;
  dateTo: string | null;
  maxResults?: number;
}

export interface AskEvidence {
  sourceId: string;
  meetingId: string;
  meetingTitle: string;
  transcriptId: string;
  snippet: string;
  context: string;
  speaker?: string;
  timestamp: string;
  audioStartTime?: number;
  audioEndTime?: number;
  meetingCreatedAt: string;
  score: number;
}

export interface AskRetrievalResponse {
  /** Retrieval identifier used by answer generation and cancellation. */
  requestId: string;
  evidence: AskEvidence[];
  provider: string;
  isExternalProvider: boolean;
}

export interface AskCitation {
  sourceId: string;
  meetingId: string;
  meetingTitle: string;
  transcriptId: string;
  snippet: string;
  speaker?: string;
  timestamp: string;
  audioStartTime?: number;
  audioEndTime?: number;
}

export interface AskClaim {
  text: string;
  citationIds: string[];
}

export interface AskAnswer {
  status: 'answered' | 'insufficient';
  answer: string;
  claims: AskClaim[];
  citations: AskCitation[];
  provider: string;
  model: string;
}

export interface AskViewState {
  phase: AskPhase;
  requestId: string | null;
  provider: string | null;
  evidenceCount: number;
  answer: AskAnswer | null;
  error: string | null;
}

File: macos/Sources/NotiveCore/Models/Ask.swift
import Foundation

public struct AskScope: Equatable, Sendable {
    public var meetingIDs: Set<String>
    public var dateFrom: Date?
    public var dateTo: Date?

    public init(
        meetingIDs: Set<String> = [],
        dateFrom: Date? = nil,
        dateTo: Date? = nil
    ) {
        self.meetingIDs = meetingIDs
        self.dateFrom = dateFrom
        self.dateTo = dateTo
    }
}

public struct AskEvidence: Identifiable, Hashable, Sendable {
    public let id: String
    public let meetingID: String
    public let meetingTitle: String
    public let transcriptID: String
    public let snippet: String
    public let context: String
    public let speaker: String?
    public let timestamp: String
    public let audioStartTime: Double?
    public let meetingCreatedAt: Date
    public let score: Double

    public init(
        id: String,
        meetingID: String,
        meetingTitle: String,
        transcriptID: String,
        snippet: String,
        context: String,
        speaker: String? = nil,
        timestamp: String,
        audioStartTime: Double? = nil,
        meetingCreatedAt: Date,
        score: Double
    ) {
        self.id = id
        self.meetingID = meetingID
        self.meetingTitle = meetingTitle
        self.transcriptID = transcriptID
        self.snippet = snippet
        self.context = context
        self.speaker = speaker
        self.timestamp = timestamp
        self.audioStartTime = audioStartTime
        self.meetingCreatedAt = meetingCreatedAt
        self.score = score
    }
}

public struct AskClaim: Identifiable, Hashable, Sendable {
    public let id: UUID
    public let text: String
    public let citationIDs: [String]

    public init(id: UUID = UUID(), text: String, citationIDs: [String]) {
        self.id = id
        self.text = text
        self.citationIDs = citationIDs
    }
}

public struct AskAnswer: Equatable, Sendable {
    public let claims: [AskClaim]
    public let citations: [AskEvidence]
    public let provider: String
    public let model: String

    public init(
        claims: [AskClaim],
        citations: [AskEvidence],
        provider: String,
        model: String
    ) {
        self.claims = claims
        self.citations = citations
        self.provider = provider
        self.model = model
    }
}

public enum AskPhase: Equatable, Sendable {
    case idle
    case retrieving
    case confirming(String)
    case generating
    case answered
    case insufficient
    case failed(String)
}
